lib/internal/errors.js

```diff
@@ -115,9 +107,37 @@ const prepareStackTrace = (globalThis, error, trace) => {
     return errorString;
   }
   return `${errorString}\n    at ${ArrayPrototypeJoin(trace, '\n    at ')}`;
-};
+}
+
+function setInternalPrepareStackTrace(callback) {
+  internalPrepareStackTrace = callback;
+}
+
+function getInternalPrepareStackTrace() {
+  return internalPrepareStackTrace;
+}
+
+/**
+ * This function is called when per-isolate PrepareStackTraceCallback is
+ * called. Every realm has its own PrepareStackTraceCallback. The callback is
+ * called with the error object created in the corresponding realm, or in VM
```

Comment:
This comment reads a bit weird to me, did you just mean something like:

> Every realm has its own PrepareStackTraceCallback. When `error.stack` is accessed, if the error is created in a shadow realm, the shadow realm's PrepareStackTraceCallback is invoked. Otherwise, the principal realm's PrepareStackTraceCallback is invoked. Note that vm contexts currently do not have their own realms, so stack generation of errors created in the vm context always uses the PrepareStackTraceCallback from the principal realm, which is shared by the main context.